### PR TITLE
Feature/5350 show solutions

### PR DESCRIPTION
**Pull request by alibombaye**
Story [AB#5350](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/5350)
Task [AB#7529](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7529) [AB#7530](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7530)

Screenshot 2020-06-05 at 15 29 33

### CODE REVIEW

**Comment by kat1906** [comment]
do you think there needs to be a bit more space below the "select catalogue solution" heading and the radio buttons?

File: app/pages/sections/catalogue-solutions/select-solution/controller.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const solutionsFound = await getData({ endpoint, accessToken, logger });
+  logger.info(`Searching for solutions for Supplier "${supplierId}" returned ${solutionsFound.length} solutions`);
+
+  return solutionsFound;
+};
+
+export const getSupplierId = async ({ orderId, accessToken }) => {
+  const endpoint = getEndpoint({ endpointLocator: 'getOrdapiSupplier', options: { orderId } });
+  const ordapiSupplierData = await getData({ endpoint, accessToken, logger });
+  logger.info(`Get selected supplier for "${orderId}" is ${ordapiSupplierData.supplierId}`);
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/select-solution/controller.js:19]: `Supplier ID for order "${orderId}" is ${ordapiSupplierData.supplierId}`

**Comment by alibombaye** [comment]
> do you think there needs to be a bit more space below the "select catalogue solution" heading and the radio buttons?

does look bit tight but that's all part of the nhs radio button component

File: app/pages/sections/catalogue-solutions/select-solution/controller.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { getContext } from './contextCreator';
+
+export const getSolutionsSelectPageContext = params => getContext(params);
+
+export const findSolutions = async ({ supplierId, accessToken }) => {
+  const endpoint = getEndpoint({ endpointLocator: 'getSolutionsForSupplier', options: { supplierId } });
+  const solutionsFound = await getData({ endpoint, accessToken, logger });
+  logger.info(`Searching for solutions for Supplier "${supplierId}" returned ${solutionsFound.length} solutions`);
+
+  return solutionsFound;
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/select-solution/controller.js:13]: I would return solutionsFound.solutions here

File: app/pages/sections/catalogue-solutions/routes.js [hunk 3]
```diff
@@ -36,13 +38,19 @@ export const catalogueSolutionsRoutes = (authProvider, addContext) => {
     return res.redirect(`${config.baseUrl}/organisation/${orderId}`);
   }));
 
-  router.get('/select', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
+  router.get('/select-solution', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
     const { orderId } = req.params;
+    const accessToken = extractAccessToken({ req, tokenType: 'access' });
+    const supplierId = await getSupplierId({ orderId, accessToken });
+    const solutionsData = await findSolutions({ supplierId, accessToken });
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/routes.js:45]: call this solutions

Comment (kat1906) [reply app/pages/sections/catalogue-solutions/routes.js:45]: (when below comment is implementedd)

File: app/pages/sections/catalogue-solutions/select-solution/controller.test.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { getSolutionsSelectPageContext, findSolutions, getSupplierId } from './controller';
+import * as contextCreator from './contextCreator';
+
+jest.mock('buying-catalogue-library');
+
+jest.mock('./contextCreator', () => ({
+  getContext: jest.fn(),
+}));
+
+describe('catalogue-solutions select controller', () => {
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/select-solution/controller.test.js:13]: select-solution

File: app/pages/sections/catalogue-solutions/select-solution/template.njk [hunk 5]
```diff
@@ -19,6 +20,12 @@
         <form method="post">
           <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
 
+          {{
+            radiobuttonOptions({
+              question: questions[0]
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/select-solution/template.njk:25]: if there is only 1 question I would get rid of the array and just pass it straight in

File: app/pages/sections/catalogue-solutions/select-solution/template.test.js [hunk 6]
```diff
@@ -60,6 +60,34 @@ describe('catalogue-solutions select page', () => {
...
+          ],
+        },
+      ],
+    };
+
+    harness.request(context, ($) => {
+      const selectSupplierRadioOptions = $('[data-test-id="question-selectSolution"]');
+      expect(selectSupplierRadioOptions.length).toEqual(1);
+      expect(selectSupplierRadioOptions.find('legend').text().trim()).toEqual(context.questions[0].mainAdvice);
+      expect(selectSupplierRadioOptions.find('input').length).toEqual(2);
```
Comment (kat1906) [inline app/pages/sections/catalogue-solutions/select-solution/template.test.js:87]: do we need test for the words beside the radio options?

File: app/pages/sections/catalogue-solutions/select-solution/template.njk (hunk 5 above)
Comment (alibombaye) [reply app/pages/sections/catalogue-solutions/select-solution/template.njk:25]: it's how we have done it in description and supplier and pretty much everywhere. I think we should leave this as is for now.

Comment (kat1906) [reply app/pages/sections/catalogue-solutions/select-solution/template.njk:25]: ok

**Pull request merged**